src/corredor/helpers/system.ts

```diff
@@ -1,14 +1,30 @@
 /* eslint-disable */
 
-import { extractID, genericPermissionUpdater, isFresh } from './shared'
-import { User } from '../../system/types/user'
-import { Role } from '../../system/types/role'
+import { extractID, genericPermissionUpdater, isFresh, Rule } from './shared'
+import { System as SystemAPI } from '../../api-clients'
+import { User, Role, Application } from '../../system/'
+
+type StrIndex<TValue> = {
+  [key: string]: TValue;
+}
+
+interface Context {
+  SystemAPI: SystemAPI;
+  $user?: User;
+  $role?: Role;
+  $application?: Application;
+}
 
 /**
  * SystemHelper provides layer over System API and utilities that simplify automation script writing
  */
 export class System {
-  constructor (ctx = {}) {
+  private SystemAPI: SystemAPI;
```

Comment:
This MUST be set, not optional.

Reply:
(same in other helpers)

Reply:
Should only the API be set or all the ctx params ?

Reply:
inspect the code and propose/decide :)

Reply:
The API and namespace params are required. Everything else is optional.

Reply:
Scratch that, namespace is optional too.